packages/sit-onyx/src/types/required.ts

```diff
@@ -0,0 +1,29 @@
+import { computed } from "vue";
+
+const REQUIRED_MARKER = ["optional", "required"] as const;
+export type RequiredMarkerType = (typeof REQUIRED_MARKER)[number];
+
+export type RequiredMarkerProp = {
+  /**
+   * Whether the form input is required.
+   */
+  required?: boolean;
+  /**
+   * Required mode: `optional` will show an `(optional)` text after the label for optional inputs.
+   * `required` will show an `*` indicator for required inputs after the label instead.
+   */
+  requiredMarker?: RequiredMarkerType;
```

Comment:
Hmm I am not sure about this property...

Reply:
If we keep it, we should define a default value of "required" in all components, otherwise it might be confusing for the users

Reply:
No, that would break the inheritance.